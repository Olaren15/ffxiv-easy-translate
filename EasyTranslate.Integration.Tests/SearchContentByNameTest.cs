using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Threading.Tasks;
using EasyTranslate.Domain.Entities;
using EasyTranslate.UseCase;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace EasyTranslate.Integration.Tests;

// Assertions are done with snapshots
[SuppressMessage("Blocker Code Smell", "S2699:Tests should include assertions")]
public class SearchContentByNameTest
{
    private readonly SearchContentByNameUseCase _searchContentByNameUseCase;

    public SearchContentByNameTest()
    {
        IServiceProvider serviceProvider = ServiceProviderBuilder.Build();
        _searchContentByNameUseCase = serviceProvider.GetService<SearchContentByNameUseCase>()!;
    }

    [Fact]
    public async Task Item_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Popoto",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Action_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "cure",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }


    [Fact]
    public async Task Achievement_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For the hoard",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task BNpcname_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "bhoot",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ClassJob_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute("summoner", Language.English);

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Companion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "lesser panda",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ContentFinderCondition_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "sastasha",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task CraftAction_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Groundwork",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Emote_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pet",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ENpcResident_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "G'raha Tia",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task EventItemLite_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Ulan's note",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Fate_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Drink Me",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Leve_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Practical Command",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Mount_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Demi-Ozma",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Orchestrion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pa-paya",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task PlaceName_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Limsa Lominsa",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Quest_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For Want Of a Memory",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Status_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Vulnerability Up",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Title_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Goddess of Magic",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Trait_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Maim and Mend",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Weather_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Fair Skies",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }
}
