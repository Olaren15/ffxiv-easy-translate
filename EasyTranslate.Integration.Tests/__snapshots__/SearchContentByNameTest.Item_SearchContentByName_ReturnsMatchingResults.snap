[
  {
    "Type": "Item",
    "IconId": 25204,
    "EnglishName": "Popoto",
    "FrenchName": "Popoto",
    "GermanName": "Toffel",
    "JapaneseName": "ポポト"
  },
  {
    "Type": "Item",
    "IconId": 27455,
    "EnglishName": "Popoto Set",
    "FrenchName": "Tubercule de popoto",
    "GermanName": "Toffel-Knollen",
    "JapaneseName": "ポポトの種芋"
  },
  {
    "Type": "Emote",
    "IconId": 64371,
    "EnglishName": "Popoto Step",
    "FrenchName": "Popping popoto",
    "GermanName": "Toffel-Tanz",
    "JapaneseName": "ポポトステップ"
  },
  {
    "Type": "Item",
    "IconId": 25204,
    "EnglishName": "Wild Popoto",
    "FrenchName": "Popoto sauvage",
    "GermanName": "Wilde Toffel",
    "JapaneseName": "ワイルドポポト"
  },
  {
    "Type": "Item",
    "IconId": 24022,
    "EnglishName": "Popoto Soba",
    "FrenchName": "Nouilles de sarrasin montagnardes",
    "GermanName": "Toffel-Soba",
    "JapaneseName": "とろろ蕎麦"
  },
  {
    "Type": "Item",
    "IconId": 24164,
    "EnglishName": "Popoto Salad",
    "FrenchName": "Salade de popotos",
    "GermanName": "Toffelsalat",
    "JapaneseName": "ポポトサラダ"
  },
  {
    "Type": "Item",
    "IconId": 25221,
    "EnglishName": "Giant Popoto",
    "FrenchName": "Popoto géante",
    "GermanName": "Riesentoffel",
    "JapaneseName": "ジャイアントポポト"
  },
  {
    "Type": "Item",
    "IconId": 53497,
    "EnglishName": "Popoto Chips",
    "FrenchName": "Popotos chips",
    "GermanName": "Frittierte Toffelscheiben",
    "JapaneseName": "ポポトチップス"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "flying popoto",
    "FrenchName": "popoto volante",
    "GermanName": "Flugtoffel",
    "JapaneseName": "フライングポポト"
  },
  {
    "Type": "Item",
    "IconId": 25204,
    "EnglishName": "Russet Popoto",
    "FrenchName": "Popoto brune",
    "GermanName": "Ockertoffel",
    "JapaneseName": "ラセットポポト"
  },
  {
    "Type": "Item",
    "IconId": 25204,
    "EnglishName": "Island Popoto",
    "FrenchName": "Popoto exotique",
    "GermanName": "Paradiestoffel",
    "JapaneseName": "アイルポポト"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "popoto peddler",
    "FrenchName": "vendeuse de popotos",
    "GermanName": "Toffel-Verkäuferin",
    "JapaneseName": "ポポト売りのペルペル族"
  },
  {
    "Type": "Item",
    "IconId": 25221,
    "EnglishName": "Dalamud Popoto",
    "FrenchName": "Popoto de Dalamud",
    "GermanName": "Dalamud-Toffel",
    "JapaneseName": "ダラガブポポト"
  },
  {
    "Type": "Item",
    "IconId": 24457,
    "EnglishName": "Mashed Popotoes",
    "FrenchName": "Purée de popotos",
    "GermanName": "Toffelbrei",
    "JapaneseName": "マッシュポポト"
  },
  {
    "Type": "Item",
    "IconId": 24072,
    "EnglishName": "Popoto Pancakes",
    "FrenchName": "Crique",
    "GermanName": "Toffelpuffer",
    "JapaneseName": "ポポトパンケーキ"
  },
  {
    "Type": "Item",
    "IconId": 25221,
    "EnglishName": "Walker's Popoto",
    "FrenchName": "Belle de Noscea",
    "GermanName": "Roter Admiral",
    "JapaneseName": "メークインポポト"
  },
  {
    "Type": "Item",
    "IconId": 25227,
    "EnglishName": "Mountain Popoto",
    "FrenchName": "Popoto des montagnes",
    "GermanName": "Bergtoffel",
    "JapaneseName": "マウンテンポポト"
  },
  {
    "Type": "Item",
    "IconId": 24163,
    "EnglishName": "Peppered Popotoes",
    "FrenchName": "Popotos au poivre",
    "GermanName": "Pfeffertoffel",
    "JapaneseName": "ペッパーポポト"
  },
  {
    "Type": "Item",
    "IconId": 23704,
    "EnglishName": "Island Popoto Set",
    "FrenchName": "Tubercules de popoto exotique",
    "GermanName": "Paradiestoffel-Knollen",
    "JapaneseName": "アイルポポトの種芋"
  },
  {
    "Type": "Quest",
    "IconId": 71221,
    "EnglishName": " Death by Popoto",
    "FrenchName": " Aux grands maux les grands remèdes",
    "GermanName": " Toffelabwehr",
    "JapaneseName": " なめずに撒いて"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "wild flying popoto",
    "FrenchName": "popoto volante primitive",
    "GermanName": "wild[a] Flugtoffel",
    "JapaneseName": "ワイルダー・フライングポポト"
  },
  {
    "Type": "Item",
    "IconId": 27456,
    "EnglishName": "Dalamud Popoto Set",
    "FrenchName": "Tubercule de popoto de Dalamud",
    "GermanName": "Dalamud-Toffelknollen",
    "JapaneseName": "ダラガブポポトの種芋"
  },
  {
    "Type": "Item",
    "IconId": 24112,
    "EnglishName": "Popotoes au Gratin",
    "FrenchName": "Gratin de popotos brunes",
    "GermanName": "Ockertoffel-Gratin",
    "JapaneseName": "ラセットポポトグラタン"
  },
  {
    "Type": "Item",
    "IconId": 24114,
    "EnglishName": "Chilled Popoto Soup",
    "FrenchName": "Soupe de popotos glacée",
    "GermanName": "Kalte Toffelsuppe",
    "JapaneseName": "冷製ポポトスープ"
  },
  {
    "Type": "Item",
    "IconId": 25236,
    "EnglishName": "Island Sweet Popoto",
    "FrenchName": "Popoto douce exotique",
    "GermanName": "Paradiessüßtoffel",
    "JapaneseName": "アイルスイートポポト"
  },
  {
    "Type": "Item",
    "IconId": 24164,
    "EnglishName": "Twilight Popoto Salad",
    "FrenchName": "Salade de popotos aux pommes",
    "GermanName": "Toffelsalat mit Apfel",
    "JapaneseName": "アップルポポトサラダ"
  },
  {
    "Type": "Item",
    "IconId": 24072,
    "EnglishName": "Giant Popoto Pancakes",
    "FrenchName": "Crique géante",
    "GermanName": "Riesentoffelpuffer",
    "JapaneseName": "ジャイアントポポトパンケーキ"
  },
  {
    "Type": "Item",
    "IconId": 24096,
    "EnglishName": "Isleworks Sweet Popoto",
    "FrenchName": "Popoto douce manufacturée",
    "GermanName": "Atelier-Süßtoffel",
    "JapaneseName": "開拓工房のスイートポポト"
  },
  {
    "Type": "Item",
    "IconId": 24164,
    "EnglishName": "Isleworks Popoto Salad",
    "FrenchName": "Salade de popotos manufacturée",
    "GermanName": "Atelier-Toffelsalat",
    "JapaneseName": "開拓工房のポポトサラダ"
  },
  {
    "Type": "Item",
    "IconId": 25453,
    "EnglishName": "Flying Popoto Repellent",
    "FrenchName": "Répulsif anti popotos volantes",
    "GermanName": "Flugtoffel-Abwehrmittel",
    "JapaneseName": "フライングポポト避けの苦汁"
  },
  {
    "Type": "Item",
    "IconId": 23741,
    "EnglishName": "Island Sweet Popoto Set",
    "FrenchName": "Tubercules de popoto douce exotique",
    "GermanName": "Paradiessüßtoffel-Knollen",
    "JapaneseName": "アイルスイートポポトの種芋"
  },
  {
    "Type": "Fate",
    "IconId": 60721,
    "EnglishName": "All Vine and No Popotoes",
    "FrenchName": "Occlusion d'otyughs",
    "GermanName": "Grenzschutz",
    "JapaneseName": "オチュー狩り"
  },
  {
    "Type": "Quest",
    "IconId": 71221,
    "EnglishName": "You Say Popoto, I Say...",
    "FrenchName": "Le bonheur est dans la forêt",
    "GermanName": "Die dolle Knolle",
    "JapaneseName": "求めるは野生種の恩恵"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "territorial flying popoto",
    "FrenchName": "popoto volante territoriale",
    "GermanName": "hinterhältig[a] Flugtoffel",
    "JapaneseName": "テリトリアル・フライングポポト"
  },
  {
    "Type": "Item",
    "IconId": 24225,
    "EnglishName": "Isleworks Sweet Popoto Pie",
    "FrenchName": "Tarte à la popoto douce exotique",
    "GermanName": "Atelier-Süßtoffeltörtchen",
    "JapaneseName": "アイル・スイートポポトタルト"
  },
  {
    "Type": "Item",
    "IconId": 24112,
    "EnglishName": "Rarefied Popotoes au Gratin",
    "FrenchName": "Gratin de popotos brunes collectionnable",
    "GermanName": "Ockertoffel-Gratin (Sammlerstück)",
    "JapaneseName": "収集用のラセットポポトグラタン"
  },
  {
    "Type": "Item",
    "IconId": 24072,
    "EnglishName": "Rarefied Giant Popoto Pancakes",
    "FrenchName": "Crique géante collectionnable",
    "GermanName": "Riesentoffelpuffer (Sammlerstück)",
    "JapaneseName": "収集用のジャイアントポポトパンケーキ"
  },
  {
    "Type": "Item",
    "IconId": 26435,
    "EnglishName": "Ballroom Etiquette - The Popoto Step",
    "FrenchName": "L'Étiquette éorzéenne : Popping popoto",
    "GermanName": "Ballsaal-Etikette: Toffel-Tanz",
    "JapaneseName": "演技教本:ポポトステップ"
  }
]
